### PR TITLE
Update to kotlin 1.5

### PR DESCRIPTION
**Pull request by smyrick**
### :pencil: Description
Update to Kotlin 1.5 and all its recommend peer dependencies

https://kotlinlang.org/docs/releases.html#release-details

### CODE REVIEW

File: plugins/client/graphql-kotlin-client-generator/src/test/data/kotlinx/custom_scalars/scalars/UUID.kt [hunk 1]
```diff
@@ -17,15 +17,15 @@ import kotlinx.serialization.json.jsonPrimitive
...
+  public val `value`: java.util.UUID
 )
 
 public class UUIDSerializer : KSerializer<UUID> {
   private val converter: UUIDScalarConverter = UUIDScalarConverter()
 
   public override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("UUID", STRING)
 
-  public override fun serialize(encoder: Encoder, value: UUID): Unit {
+  public override fun serialize(encoder: Encoder, `value`: UUID): Unit {
```
[the remark this answers is not in the document]
Comment (dariuszkuc) [reply plugins/client/graphql-kotlin-client-generator/src/test/data/kotlinx/custom_scalars/scalars/UUID.kt:28]: could be tackled in separate PR

**Review by dariuszkuc — COMMENTED** [review]
Overall looks fine but can you check whether the lib will work with projects using Kotlin 1.3 and/or 1.4? (e.g. https://youtrack.jetbrains.com/issue/KT-40226)

**Comment by smyrick** [comment]
@dariuszkuc I updated our internal SDK to use the latest snapshot version compile with kotlin 1.5. The SDK uses kotlin 1.4.32.

I was able to start the example app, as well as run the maven plugin. I bet we could test the local graphql-kotlin-examples running Kotlin 1.4 but using the lib with 1.5 to test the gradle plugin.

That issue you link to does rely on the JS runtime which we do not use so it may be JS Gradle specific

**Comment by smyrick** [comment]
I updated `graphql-kotlin-example` to use the following and everything looks good

```kt
val kotlinVersion = "1.4.32"
val kotlinCoroutinesVersion = "1.4.3"
```

**Review by dariuszkuc — APPROVED** [review]

**Pull request merged**